src/gt4py/next/otf/compiled_program.py

```diff
@@ -253,7 +260,9 @@ def __call__(
             raise RuntimeError("No program compiled for this set of static arguments.") from e
 
     @functools.cached_property
-    def _argument_descriptor_cache_key_from_args(self) -> Callable:
+    def _argument_descriptor_cache_key_from_args(
+        self,
+    ) -> Callable[..., tuple[ScalarOrTupleOfScalars, ...]]:
```

Comment:
```suggestion
    def _argument_descriptor_cache_key_from_args(
        self,
    ) -> Callable:
```
The output type depends on what the argument descriptor extracts and can be anything. In #2173 this consequently changes to also include domains.

Reply:
Funny thing is that I didn't intend to make this change, but Copilot suggested it to fix typing issues.